### DISCUSSION TITLE
MPD does not start with NAS mounted

### DISCUSSION THREAD

**Discussion by Lubos-V** [rern/rAudio#1388] [Bug Reports]
Hi @rern,

I wanted to listen to some music and realized today that the UI is not responsive (loading indefinitely, displaying the logo). Tried restart, command line update to latest published version, then command line update to the last development version - still to no avail.

systemctl reports mpd daemon failing with
decoder: Decoder plugin 'wildmidi' is unavailable: configuration file does not exist:

journalctl delivers the same

Any idea what's wrong? I haven't change the config recently...

**edit:**
**after analysis, the problem is different - MPD fails to start when my NAS is mounted. unmounting it, starting MPD, and remounting works. still, no idea what is wrong...**

**Comment by Lubos-V** [comment]
... @rern to add to the observations, updates actually fail after downloading the files with (obvious?) error:

```
Job for mpd.service failed because a timeout was exceeded.
See "systemctl status mpd.service" and "journalctl -xeu mpd.service" for details.
MPD error: Connection refused
MPD error: Connection refused
MPD error: Connection refused
/srv/http/bash/status.sh: line 163: ((: > 0 : syntax error: operand expected (error token is "> 0 ")
mpdnotrunning
```

**Reply by klassikmann** [reply]
@Lubos-V 

Personally I have no experience with wildmidi in mpd but do you specify a config_file in the corresponding deocder plugin of mpd.conf as described in the mpd documentation?

```
decoder {
    plugin "wildmidi"
    config_file "/etc/timidity/timidity.cfg"
}
```
Another idea: Look at wildmidi issues in the github of mpd:
[https://github.com/MusicPlayerDaemon/MPD/issues?q=is%3Aissue+in%3Atitle+wildmidi](url)

**Reply by Lubos-V** [reply]
@klassikmann I haven't made any changes to the standard configuration and am certainly not using wildmidi intentionally. My raudio installation stopped "working" out of nowhere and I am somewhat stuck at the moment...

**Reply by klassikmann** [reply]
@Lubos-V 
Have a look into all \*.conf files in `/srv/http/data/mpdconf` to find a reference to the wildmidi decoder.

**Reply by Lubos-V** [reply]
@klassikmann grep doesn't provide anything usable within that folder...

**Comment by ceejayemm** [comment]
The 'wildmidi' configuration has been missing for some  time.  I get the same error message (via journalctl -xe) but rAudio works as normal and mpd functions as it should for me:

-- A start job for unit mpd.service has begun execution.
--
-- The job identifier is 5707.
Jul 07 10:40:41 rAudio-1 mpd[35373]: decoder: Decoder plugin 'wildmidi' is unavailable: configuration file does not exist: /etc/timidity/timidity.cfg
Jul 07 10:40:41 rAudio-1 systemd[1]: Started Music Player Daemon.
-- Subject: A start job for unit mpd.service has finished successfully

The missing config file is noted (back in 2019) here:

https://github.com/MusicPlayerDaemon/MPD/issues/589

I believe you can simply create a blank /etc/timidity/timidity.cfg  file to get rid of this spurious error message.

As for the actual problem you are experiencing with MPD not starting I can't help with this other than to say that I have no such issue with this (but do have others) with the 20230630 update.

Chris

**Reply by Lubos-V** [reply]
that was a good hint @ceejayemm 

I dug a little deeper, also looking at the script referenced in the error message

`/srv/http/bash/status.sh: line 163: ((: > 0 : syntax error: operand expected (error token is "> 0 ")`

likely this is failing

`songpos=$( mpc status %songpos%)`

leading to the error shown above (as likely songpos is empty)

`(( $songpos > 0 )) && song=$(( songpos - 1 )) || song=0 # mpd song    : start at 0`

executing mpc status gives me MPD connection refused

still, no idea what is really going wrong and why...

**Comment by ceejayemm** [comment]
Lubos-V

I would be more concerned about the repeated error:

MPD error: Connection refused

as this might be the reason why the final (songpos) error is being generated.

Which version of rAudio are you using ?  Where is your music Library (device /file system) ? What permissions are  on the library files ?

**Reply by Lubos-V** [reply]
@ceejayemm:

music library is on a NAS (have used in for quite some time now); permissions 755, user/group mpd/mpd; can 'head' the individual files and browse the library on command line



fstab entry is as follows:
//192.168.178.1/FritzNAS/NAS/music  /mnt/MPD/NAS/FritzNAS  cifs  ro,noauto,noserverino,username=xxx,password=xxx,uid=45,gid=45,iocharset=utf8  0  0

this is the latest version of raudio; as mentioned i also tried to run the update (to both final and beta) from the command line - to no avail. files get downloaded, then the process "freezes" for a minute or so (trying to start MPD i guess) and then it fails

rebooting raudio proceeds fine - the screen attached shows UI in the background with the "raudio loading" overlay icon which remains displayed indefinitely. the same on webbrowsers.

**Comment by ceejayemm** [comment]
Ok, any reason why you are updating from command line, what do you mean by 'final' and 'beta' ?  Have you tried updating from the Cog menu / Addons / Update ?  Does this succeed? Why is you library mounted with the 'ro,noserverino' options?  Did you mount it via the System page ? At the top of the System page what version is displayed ?

Chris

**Reply by Lubos-V** [reply]
@ceejayemm I can currently only work via the command line as the "cog menu" is not accessible in the browser - the UI is blocked by the pulsating raudio icon overlay and I can do nothing about it.

final should read stable
```
curl -sfLO https://github.com/rern/rAudio/raw/main/install.sh
chmod 755 install.sh
./install.sh "r1
Update
main"
```

beta is the bleeding edge
```
curl -sfLO https://github.com/rern/rAudio/raw/UPDATE/install.sh
chmod 755 install.sh
./install.sh "r1
Update
UPDATE"
```

the library is mounted ro as there is no need why raudio should be able to write onto the share; noserverino was needed in order to be able to mount the network share (without this parameter, mounting failed), and yes, if I remember correctly I mounted it via the user interface (while providing the necessary parameters).

It has worked for the last couple of months. I recall doing a normal update via the user interface a couple of days ago. On Saturday it still worked, yesterday not anymore. After that I tried the command line updates and a couple of restarts to no avail.

**Reply by Lubos-V** [reply]
@ceejayemm hmmm, it seems that on my other computer, I can "skip" the overlay and get to the UI. 

version displayed is rAudio 20230630
drive seems to be mounted (green light), the library shows total number of tracks but NAS entry seems to be empty and cannot be "rescanned".
and, the MPD screen indicates "MPD is not running"

**Reply by ceejayemm** [reply]
I am not sure what you mean by 'skip the overlay', from a web browser accessing the system IP number should take you straight to the Playback main screen.  From there you should be able to access all other pages, including the Setup (Cog) menus.

 I'm out of ideas now.  The only thing I would do in this situation is to reflash the SD card, reconfigure the system and start again.

1. reflash the card with the latest version from the RAudio github site
2. mount the NAS share via the system page
3. Configure any dac/sound cards/bt speakers etc
4. Set the ffmpeg setting in Player (if needed)
5. Set any Features needed (may be only the absolutely necessary options to start with?)
6. Rebuild the database using the Full Rebuild option

See what happens then.

Chris

**Reply by klassikmann** [reply]
I second Chris' proposal with one amendment: Forget the UPDATE branch.

**Reply by Lubos-V** [reply]
@klassikmann @ceejayemm I will indeed likely have to reflash - it's a good exercise to do from time to time anyway.

Regarding the overlay: when system is loading / working, the screen displays a pulsating raudio icon on the screen but also on the web front-end. I explored a bit - depending on the browser (iPhone is particularly bad), it's possible to click on the screen and get rid of this overlay icon (on iPhone it's not possible). After that, the UI is accessible again (but whatever I do, no positive effect, MPD doesn't start, MPC fails connecting to it, ...). Thanks for the ideas anyway, gentlemen!

**Reply by Lubos-V** [reply]
@ceejayemm your direction was spot on. after getting back to the system, I have done the following:
- unmount the file share
- restart MPD (via command line)
after that, MPD starts immediately and I can play webradios.

that means - what's wrong about the settings of the NAS share?

after re-mounting using the same settings, i can then start playback of audio from the share.

after reboot, the problem emerges again. so it's something preventing the mounting of the share properly (even though the config is obviously OK and I can access the data - not sure whether MPD can)

puzzling...

**Reply by Lubos-V** [reply]
more infos @ceejayemm :

after reboot, mount reports
`//192.168.178.1/FritzNAS/NAS/music on /mnt/MPD/NAS/FritzNAS type cifs (ro,relatime,vers=3.1.1,cache=strict,username=XXXXX,uid=45,noforceuid,gid=45,noforcegid,addr=192.168.178.1,file_mode=0755,dir_mode=0755,iocharset=utf8,soft,nounix,mapposix,rsize=65536,wsize=65536,bsize=1048576,echo_interval=60,actimeo=1)`

and it stopped working again.

the only way to make things work is (this is reproducible)

- unmount share via UI
- start the daemon
- remount the share via UI**

after that, playback is possible without problems. the mount entry looks the same.

so, any idea what MPD doesn't like here / what is going on?

btw, where are the mount options coming from? if i specify "ro" for mount, where is the file_mode and dir_more coming from?

**Reply by ceejayemm** [reply]
Are you mounting the share via the web app or directly via fstab?  I have never had any problem mounting shares using the web app (but I typically use NFS not CIFS shares) and so do not edit fstab directly. Again I would question the need for the extra options in the mount command, in fact you now seem to have even more than before. I would try mounting the share from the web app with NO Options selected and then ONLY add the options one at a time, but only if really needed. I know you are trying to preserve the security of your NAS but it might be worth considering relaxing these requirements a bit, for a short time, until you know which options will work and which give you problems.

Chris

**Reply by Lubos-V** [reply]
@ceejayemm I am mounting it via the web interface - that's the strange thing. It's just "if mounted on boot, MPD will fail".

As root, I can access the data without any problems via the command line. just su'd myself as mpd. can access the data...

The options there are not coming from my fstab (and I am wondering, where they are coming from) - ro,noauto,noserverino,username=xxx,password=xxx,uid=45,gid=45,iocharset=utf8 0 0
are the ones I am having.

The mount point is mpd/audio before mounting and mpd/mpd after mounting but that shouldn't make any difference, I think...

Strange... will continue experimenting tomorrow...

**Reply by Lubos-V** [reply]
@ceejayemm and one more thing - this exercise might not be successful. If the MPD is already running, I can imagine the mounting will work (as in the current case). Maybe 1) mount 2) restart MPD 3) check will be needed...

**Reply by ceejayemm** [reply]
A 'standard' CIFS mount via the web app for me, with no Options selected,  

Name: Test_Music
Server IP: 192.168.86.250
Share: music
User: xxx
Password: xxx
Options: <none>

gives:

//192.168.86.250/music                     /mnt/MPD/NAS/Test_Music  cifs  noauto,username=xxx,password=xxx,uid=45,gid=45,iocharset=utf8    0  0

the Share mounts as /mnt/MPD/NAS/Test_Music and has mpd/mpd and 755 ownership/permissions

Chris

**Reply by klassikmann** [reply]
@Lubos-V
I'm using the same way of "standard" CIFS mounting with the web app, with the exact same positive result.
My advice: Stop the "out-of-GUI" treatment of shares, mpd restarts and stuff, that's the root cause of your problems (IMHO).
Use the GUI exclusively.

**Reply by Lubos-V** [reply]
@ceejayemm  @klassikmann  Now I know why I used "noserverino". If I don't, I am getting endless update of the library and "stale file handle" when trying to access individual files mounted from the NAS. Entered everything via the UI again, including the noserverino parameter - fstab looks the same (with noserverino parameter), and the output of mount as well (with the much longer input obviously triggered by the noserverino parameter).

need to run the library update (it takes a long time) and check whether restarting the MPD or the whole system is going to have the same problems - if it does, that would be an indication for a bug for me...

**Reply by Lubos-V** [reply]
@ceejayemm  @klassikmann  @rern 
the problem is confirmed - for CIFS share, i need to put noserverino to the mount list to make it work and (re-)generate the library. after reboot MPD fails to start. need to (doing this from the UI) unmount the share, start the MPD, and the re-mount the share. after that, everyting works perfectly (until the next reboot).

this is really strange...

**Reply by kw-sudo** [reply]
Hi Lubos-V,

my settings are identical, focus on bitperfect audio (within r/Audio config).

To shorten it, currently I can add my NAS via setting form. Thereupon r/Audio added this string to fstab:

`//192.168.7.5/music   /mnt/MPD/NAS/DATENSERVER  cifs  noauto,username=xxxxxx,password=xxxxxxxxxxxxx,uid=45,gid=45,iocharset=utf8  0  0`

‚music‘ is the standard Synology media library folder, no special permissions. Works. :-)

**Reply by Lubos-V** [reply]
@master-kw 

yes, this _normally_ works. for me, it worked like this for the last 5 or so years. some time ago, I switched from a dedicated NAS server to a different NAS approach - this time done via a drive connected directly to the FritzBox router serving as NAS.

in order to make this setup work (maybe there are some quirks in the support of the CIFS protocol), _noserverino_ has to be added to the list of mount parameters (otherwise I am getting _stale file handle_ errors for the share mounted on my rAudio). please note that the share is perfectly accessible and being used from a couple other devices.

this parameter (or something related to it) seems to conflict with something on rAudio as having the share mounted by the processes after reboot prevents MPD from starting. the workaround is to unmount (in the UI), start MPD (in the UI), remount (in the UI) - then playback is possible without any problems.

**Reply by ceejayemm** [reply]
Ok, now we are getting somewhere. My fault I should have asked earlier what you were using a your 'NAS'.  It seems there are documented 'problems' with using the FritzBox CIFS as a 'NAS' which entailing using the 'noserverinfo' option to get around them.  Do a Google search for 'fritzbox cifs mount' and see what you get. I see references to SMB v1.x, stale file handler, noserverinfo etc etc - all of which you have mentioned and more.

If possible I would consider replacing your Fritzbox 'NAS' with something else which has a better CIFS implementation as it looks to me like this is the cause of your issues.  I know you say you have used it successfully in the past but it now seems obvious that the current variant is probably the root cause of your issues.

Chris

**Reply by R101G** [reply]
@Lubos-V 
Early versions of Samba do get deprecated, and most routers do not have the CPU power even for v3, so I assume you have tried vers=2.0 in the options?

**Reply by Lubos-V** [reply]
hi @R101G the router explicitly states support for v3 (v1 has to be activated extra). v2 didn't work (stale file handle also there).

something might be wrong about the way how Fritz implemented the protocol (as @ceejayemm suggests).

might need to live with the workarounds (not rebooting raudio frequently) and think about maybe setting new fileserver on additional raspi...

**Reply by R101G** [reply]
@Lubos-V 
A change of server does seem the simplest solution. I use a Pi running Raspian Lite as a CIFS server without any problems.

**Reply by Lubos-V** [reply]
Looks like that @R101G. Just installed OpenMediaVault on my other raspi. Had to wipe my NTFS-based external drive and reformat it, but the USB3 is quite fast so it just took under two hours to restore the 300+GB of data from another backup drive.

We haven't fixed the original problem but maybe this can help people if they encounter similar problem.

**RAudio, FritzBox, NTFS share, mount noserverino, MPD fails to start on reboot.**

**Comment by Lubos-V** [comment]
... I guess there is not much more to be added. Closing the thread. Thanks for your valuable hints, gentlemen!

**Comment by marcelgsteiger** [comment]
my late 2 ¢: I am connecting from my Raspi 4/4GB raudio player using SMB to my music library as follows:

- NAS: FritzBox 7590 w/Firmware 7.56 (7.50 worked too)
- external USB SSD SanDisk 1TB **reformatted with ext4**
- Wifi 2.4 / 5GHz

connection options in the raudio mount dialog (multiple raudios connecting to the same SMB share):

`ro,uid=1000,gid=1000,vers=3.0,noserverino`

This avoids the NTFS overhead on the FritzBox. Moreover, I'm forcing protocol version down to 3.0 which appears to be sufficient and more appropriate here than later versions. To decouple the player from the network, I allow mpd to pre-buffer entire tracks by adding a user's configuration for mpd:

`input_cache {`
`   size "2 GB"`
`}`

Few seconds before a track ends, mpd then starts to pre-buffer the entire next track. This works well even with DSD128 tracks.
HTH anyone.

Regards
Marcel

**Reply by Lubos-V** [reply]
@marcelgsteiger thanks for additional input. I actually complete forgot that FB supports ext4 as well. Have now a dedicated (rpi-based) NAS solution but will experiment further with FB as it is an interesting alternative.

do you need to put the input_cache explicitly into the MPD configuration? Isn't this what the buffer setting in the raudio GUI is for? (don't have access at the moment and cannot thus tell you the exact place / name)

**Reply by marcelgsteiger** [reply]
@Lubos-V The buffer setting in the raudio GUI appears to work differently. To be honest, I do not know which is better. I observe mpd buffering using
`watch ps Hh -p $(pidof -s mpd) -o 'pid,tid,class,priority,comm,s,pcpu,vsize,drs'`
buffering entire 44kHz PCM tracks typically only takes a few seconds and usually finishes before the track begins to play. DSD128 takes somewhat longer, of course. This mpd buffering makes me hope that a track can finish playing completely even when the network has an outage. As of now, I don't have tested this thoroughly yet.
